File: programmers/python 3/12905.py
# 문제 링크: https://programmers.co.kr/learn/courses/30/lessons/12905

def solution(board):
    if not max(board[0]) and not max(row[0] for row in board):
        return 0

    max_side = 1
    for i in range(1, len(board)):
        for j in range(1, len(board[0])):
            if not board[i][j]:
                continue
            board[i][j] = min(
                board[i - 1][j],
                board[i][j - 1],
                board[i - 1][j - 1]
            ) + 1
            max_side = max(board[i][j], max_side)

    return max_side**2

File: programmers/swift/12905.swift
// 문제 링크: https://programmers.co.kr/learn/courses/30/lessons/12905

func solution(_ board:[[Int]]) -> Int {
    guard board[0].max() == 1 || board.map({ $0[0] }).max() == 1 else {
        return 0
    }

    var board = board
    var maxSide = 1
    for i in 1 ..< board.count {
        for j in 1 ..< board[0].count {
            guard board[i][j] == 1 else { continue }

            board[i][j] = min(
                board[i - 1][j],
                board[i][j - 1],
                board[i - 1][j - 1]
            ) + 1

            if board[i][j] > maxSide {
                maxSide = board[i][j]
            }
        }
    }

    return maxSide * maxSide
}